PEP8 compliance update

thecsw opened the pull request:
Hey! So I thought about our code. It is pythonic but after looking at it after a long break, I found some sinister things in it:

- HUGE try-except blocks
- HUGO main functions with >50 statements
- Lots of `for`s and `if`s in the code. Too many branches
- We don't have proper function documentation
- Constant variables are not uppercase
- more sins

I worked on making the code more compliant with [PEP8 style](https://www.python.org/dev/peps/pep-0008/), the official python3 ceding style.  I reworked most of the boring things. Un-spaghettified the `submitter.py`, got rid of the huge try block. Made some helper functions. I need to do the some with main.py. It is Thanksgiving Day and I have to go in 20 mins, I will finish this later. It's nearly done, I tested some parts of the code, it works fine. 

closes #128

Happy Thanksgiving!

thecsw committed:
Made general optimizations and refactored submitter.py

thecsw committed:
Made everything more compatible with PEP8

https://www.python.org/dev/peps/pep-0008/

thecsw committed:
Added no warn script to the dockerfile

thecsw committed:
Reduced the number of statements and branches in main

PEP8 Compliance

thecsw committed:
Added docstrings to functions and imports

thecsw committed:
Added docstrings to functions

convention (C0111, missing-docstring, investments_post) Missing function docstring

thecsw committed:
Made some updates. Needs final polishing

I would like to finish this but it's Thanksgiving Day and I'm going to a Thanksgiving dinner now, will finish later, thanks!

thecsw committed:
Made more updates, huh?

thecsw commented:
This update will close #128 
The oldest issue, wow

thecsw committed:
Made more PEP8 compliance updates

thecsw committed:
Added more docstrings, todos

thecsw committed:
Cleaned up the calculator module

thecsw committed:
Fixed the submitter bug with database and added more todos

thecsw committed:
Added sleeping for database

thecsw committed:
Fixed custom time formatting

TwiN commented on src/api.py:
```diff
@@ -331,8 +371,12 @@ def index():
+    """
+    !!! DEPRECATED !!!
+    Calculate investment
+    """
```
Don't you feel that it might be worth considering adding a decorator for deprecated methods?

A good example would be https://deprecation.readthedocs.io/en/latest/

Or if we don't want to add an external package for a deprecation decorator, we could always create our own.
It may not be _very_ useful currently, but as the project grows, it'll definitely help making the code more maintainable.

thecsw replied:
That's a really good idea!

thecsw committed:
Cleared all the trailing whitespaces

thecsw committed:
I tried to make a hardcoded binding. Issue due to codacy review

Binding to all network interfaces can potentially open up a service to traffic on unintended interfaces, that may not be properly documented or secured.

thecsw commented:
Merging this. It will take a couple of days to test it.

thecsw commented:
The deprecated option will come in another PR.
